File: Cpp/190_Reverse_Bits.cpp
class Solution {
public:
    uint32_t reverseBits(uint32_t n) {
        bitset<32> b = n;
        int bit_size = b.size();
        for(int i=0; i<bit_size/2; ++i){
            int t = b[i];
            b[i] = b[bit_size-1-i];
            b[bit_size-1-i] = t;
        }
        
        return b.to_ulong();
    }
};

File: Kotlin/190_Reverse_Bits.kt
class Solution {
    // you need treat n as an unsigned value
    fun reverseBits(n:Int):Int {
        var result = 0
        
        for (i in 0 until 32) {
            result = (result shl 1) or ((n ushr i) and 1)
        }
        
        return result
    }
}